src/librustc_mir/transform/const_prop.rs

```diff
@@ -619,8 +619,19 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
         }
     }
 
-    fn should_const_prop(&self) -> bool {
-        self.tcx.sess.opts.debugging_opts.mir_opt_level >= 2
+    fn should_const_prop(&mut self, op: OpTy<'tcx>) -> bool {
+        if self.tcx.sess.opts.debugging_opts.mir_opt_level >= 2 {
+            return true;
+        }
+
+        match *op {
+            interpret::Operand::Immediate(Immediate::Scalar(ScalarMaybeUndef::Scalar(s))) =>
+                s.is_bits(),
+            interpret::Operand::Immediate(Immediate::ScalarPair(ScalarMaybeUndef::Scalar(l),
+                                                                ScalarMaybeUndef::Scalar(r))) =>
+                l.is_bits() && r.is_bits(),
+            _ => false
+        }
```

Comment:
Maybe lower the minimal mir opt level for const prop instead, so you can still disable it?

Reply:
That would only be available to nightly users, so it doesn't seem too useful to me. Or is mir-opt-level=1 active in debug mode?

Reply:
We could use 1 as default mir opt level and move all opts which are currently enabled on opt level 1+ one opt level up.

Reply:
1 is the default, independent of llvm optimization level.